### ISSUE TITLE
Revise `weeklyArena` GQL Query

### ISSUE THREAD

**Issue by longfin** [planetarium/NineChronicles.Standalone#1414]
After [v100230][], we introduced a new areana system. but `weeklyArena` query wasn't updated and it's still returning based outdated arena scheme.

[v100230]: https://wiki.nine-chronicles.com/en/9C/patch-notes#v100230

**Comment by TarogStar** [comment]
I have been working on an update for this, and I have a couple of questions.  The ArenaInformation returned has the Ticket property set to zero unless the avatar has used at least 1 ticket after the reset.  I'm guessing that I need to look at the TicketResetCount and use ArenaHelper.GetCurrentTicketResetCount and compare to decide if they should have zero or 8 tickets available when the value returns as zero.  That does require that I am able to get the current block index, and within the StateContext of StateQuery I don't see an easy way to get that information.

**Comment by longfin** [comment]
> That does require that I am able to get the current block index, and within the StateContext of StateQuery I don't see an easy way to get that information.

I think that we can add the current block index to the `StateContext`. 

@moreal @ipdae or, already there is?

**Comment by TarogStar** [comment]
I also needed the name and agent address.  I didn't really want to slow it down by getting the entire avatar state, but I can if that is expected.  I used the following code to get just that information: 
`if (context.Source.GetState(participantAddress) is Dictionary participantState)
                        {
                            if (participantState.ContainsKey(Lib9c.SerializeKeys.LegacyNameKey))
                            {
                                arenaInfo.AgentAddress = participantState[Lib9c.SerializeKeys.LegacyAgentAddressKey].ToAddress();
                                arenaInfo.AvatarName = participantState[Lib9c.SerializeKeys.LegacyNameKey].ToDotnetString();
                            }
                            else
                            {
                                arenaInfo.AgentAddress = participantState[Lib9c.SerializeKeys.AgentAddressKey].ToAddress();
                                arenaInfo.AvatarName = participantState[Lib9c.SerializeKeys.NameKey].ToDotnetString();
                            }
                        }`

**Comment by ipdae** [comment]
>That does require that I am able to get the current block index, and within the StateContext of StateQuery I don't see an easy way to get that information.

  Yes. `stateQuery` already use current block index. and you can configure specific block hash like this.

```
query {
  stateQuery(hash: ) {
    avatar(avatarAddress: "0x709e5bf0823df4ddff76a11a6585d1463e9dc9f8") {
      eventMap {
        pairs
      }
    }
  }
}
```

**Comment by ipdae** [comment]
>If it is a common request to only get certain fields, would you like me to write a function that can selectively get the properties of the avatar to make those requests that don't require Inventory/Quest/World Information?

currently, inventory, quest, worldInformation already seperate each address. so, we can add avatar query with optionally fields.

**Comment by TarogStar** [comment]
> Yes. `stateQuery` already use current block index. and you can configure specific block hash like this.
I submitted a PR for my current code. I do see the chain tip index in the StandaloneQuery which calls the StateQuery, so I added a way to pass the actual index of the chain tip, which is what is currently required to be able to compare with the start and end block indexes.  Let me know if I should use a different method.
